scripts/setup.py

```diff
@@ -52,23 +54,16 @@ def main():
 
     logging.basicConfig(format='%(levelname)s: %(message)s')
 
-    source_root = util.read_source_root_path()
-
-    # the script is being run in the cbmc root
-    cbmc_root = os.path.abspath('.')
-
-    # the script is creating the proof root
-    proof_root = os.path.abspath('proofs')
-
-    # the script is linking to the litani script within the litani submodule
-    litani = util.read_litani_path()
-
-    # the name of the project used in project verification reports
-    project_name = util.read_project_name()
+    cbmc_root = Path.cwd()
+    proof_root = cbmc_root / "proofs"
+    source_root = repository.repository_root()
+    litani = repository.litani_root() / 'litani'
```

Comment:
`submodule_root` concerns me because it raises on failure. In future commits, we plan to use the Litani submodule only if it already is in the repo, and to check for Litani on the PATH otherwise. But that doesn't work if the method throws on failure.

I guess eventually the code will look like this?
```
try:
   litani = repository.litani_root() / "litani"
catch UserWarning:
    litani = os.which("litani") or raise UserWarning("no submodule and no path")
```
?

Reply:
I've changed `submodule_root` to return None in light of your prior observations.